### PR TITLE
Update instead of adding multiple key:value props

### PR DESCRIPTION
**Pull request by hjanetzek**
- Renamed add to set

Thought it would make a good PR on its own :)

### CODE REVIEW

**Comment by tallytalwar** [comment]
:+1:

**Comment by tallytalwar** [comment]
This is a breaking change on eraser-map builds. 
So, #489 and #491 should be merged in parallel and then https://github.com/mapzen/eraser-map/pull/264 on eraser-map.

File: core/src/data/properties.cpp [hunk 1]
```diff
@@ -50,6 +50,8 @@ const Value& Properties::get(const std::string& key) const {
     return it->value;
 }
 
+
+
```
Comment (blair1618) [inline core/src/data/properties.cpp]: ?
?

Comment (blair1618) [inline core/src/data/properties.cpp]: :)

**Comment by blair1618** [comment]
Looks good to me. Let's merge this!

This could substantially improve parse time for JSON tile formats - right now feature properties are collected by making repeated calls to `Properties::add()` which is needlessly re-sorting the list over and over.

**Comment by blair1618** [comment]
Not necessarily a concern for this PR, but [this bit in Properties](https://github.com/tangrams/tangram-es/blob/master/core/src/data/properties.cpp#L12-L18) seems like it can (should) be replaced with `std::swap(props, _items)`

**Comment by tallytalwar** [comment]
Agreed!

**Comment by hjanetzek** [comment]
This constructor was used specifically for mvt source. The idea was to reuse the allocated vector on the caller site and just clobber the values from it.

**Comment by tallytalwar** [comment]
Rather `props.swap(_items)`

**Comment by blair1618** [comment]
Ah, ok. I'll leave that constructor alone for now. I can do what I wanted with the slightly uglier
```
properties.setSorted(unsortedItems);
properties.sort();
```

**Comment by tallytalwar** [comment]
Can we do this outside this PR?

**Comment by blair1618** [comment]
Yeah I'm just thinking about changes elsewhere. Carry on!

**Pull request merged**